### DISCUSSION TITLE
How to find out which installed Apps can be updated by winget and which cannot (show which ones are available in the repo)

### DISCUSSION THREAD

**Discussion by rwiesbach** [microsoft/winget-cli#3624] [Q&A]
I have a Windows that did not use winget before and has a bunch of tools installed. How can I find out which tools can be updated by winget and which cannot because they are not included in the repo(s) ? (e.g. which I do have to suggest to the repo or keep track of manually)

`winget list --source winget`

definitely shows a) applications that have not been installed via winget and b) applications that are not available in the winget repo (verified by 'winget search %id%') and c) the column "available" is only filled if a newer version is available (it is empty if the same version is available and if the application is not available in the repo

`winget list --upgrade-available`

shows apps that are in the repo (as the repo has updates for them) but it (obviously) does not show applications that are in the winget repo but up-to-date.

**Comment by Trenly** [comment]
If you use `winget list` you should see there is an `Id` column. This column will be one of two things. If the package is available in a configured source, it will be the package identifier -for example `Github.Cli` or `Microsoft.Edge`. If the package is not available, then it will be the system reference string (either the product code or package family name) - for example `{00001220-0220-1033-84C8-B8D95FA3C8C3}` or `Microsoft.WindowsAlarms_8wekyb3d8bbwe`.

The other option is to use `winget export -o <path-to-output-file>` which will provide a json list of all the packages and their source and will output to console the list of packages not available from any source

**Comment by rwiesbach** [comment]
Thanks. I was not aware that GUID and _%hexSomething% suffixes are equivalent to applications unknown to winget.
